src/Vlingo.Actors/Plugin/Mailbox/AgronaMPSCArrayQueue/ManyToOneConcurrentArrayQueueDispatcher.cs

```diff
@@ -32,27 +33,36 @@ public class ManyToOneConcurrentArrayQueueDispatcher : IRunnable, IDispatcher
             Mailbox = new ManyToOneConcurrentArrayQueueMailbox(this, mailboxSize, totalSendRetries);
             this.throttlingCount = throttlingCount;
             closed = new AtomicBoolean(false);
-            cancellationTokenSource = new CancellationTokenSource();
+            dispatcherTokenSource = new CancellationTokenSource();
+            backoffTokenSource = CancellationTokenSource.CreateLinkedTokenSource(dispatcherTokenSource.Token);
         }
 
-        public void Close() => closed.Set(true);
+        public void Close()
+        {
+            closed.Set(true);
+            dispatcherTokenSource.Cancel();
+            dispatcherTokenSource.Dispose();
+            backoffTokenSource.Dispose();
+        }
 
-        public bool IsClosed => closed.Get() || cancellationTokenSource.IsCancellationRequested;
+        public bool IsClosed => closed.Get();
 
         public void Execute(IMailbox mailbox)
         {
-            cancellationTokenSource.Cancel();
+            backoffTokenSource.Cancel();
+            backoffTokenSource.Dispose();
+            backoffTokenSource = CancellationTokenSource.CreateLinkedTokenSource(dispatcherTokenSource.Token);
         }
 
         public bool RequiresExecutionNotification { get; }
 
-        public void Run()
+        public async void Run()
```

Comment:
Using `async void` here seems more like a hack. I would disagree its use here for few reasons. 

- `async void` really should only be used in very top level methods where you cannot change the signature but need to use `await` inside the method body. This does not seem to such a place.
- We have a plan to use the Task with async/await fully later (in distant future) which may propagate the task all the way to the top. However, this usage stops propagating Task at this level. It also hides the fact that Task has been used to implement some requirements in the levels below. 

I, however, understand why you needed to do it. This seems like the easiest way to get rid of the `Thread.Sleep(...)` calls in Backoff. If we really must do it as a temporary measure, I would really like to do it in the Backoff class. That is, `Backoff` class method should not return Tasks, rather they become `async void` methods and await on the Task inside the method body. We will replace them later (again, in distant future) when we use Tasks everywhere.

Reply:
Hi @zpbappi. Of course this is a temporary mesure to fix quickly #28. You do well to point this out so I'm happy we'are on the same page. Even if this is not the final implementation I do think it is still better than `Thread.Interrupt` which may not only blow at the place where sleep was, but accidentally in another place that we don't control. So if you agree I would merge that and quickly go towards full implementation of Tasks getting rid of those quicks wins.

Reply:
Ok. Let's keep the change that you've made. I am creating an issue to address it later. In terms of behavior, it should be correct.